File: dangerfile.js
// dangerfile.js
const { danger, fail, warn, message } = require("danger");

// Check for changes in test files
const testFilesChanged = danger.git.modified_files
  .concat(danger.git.created_files)
  .some(file => file.includes("test"));

if (!testFilesChanged) {
  warn("This PR doesn’t appear to include any test changes. Please consider adding or updating tests.");
} else {
  message("Thanks for adding unit tests")
}

File: dangerfile.dart
import 'package:danger_core/danger_core.dart';

const bigPRThreshold = 300;

void main() {
  // WIP check
  if (danger.github.pr.title.contains('WIP')) {
    warn('PR is considered WIP');
  }

  // Unit tests check
  final allSourceFiles = [
    ...danger.git.modifiedFiles,
    ...danger.git.createdFiles,
  ];

  if (allSourceFiles.any((file) => file.endsWith('_test.dart'))) {
    message('Thanks for updating the unit tests');
  }

  // Big PR check
  final prAdditions = danger.github.pr.additions ?? 0;
  final prDeletions = danger.github.pr.additions ?? 0;
  final isBigPR = (prAdditions + prDeletions) > bigPRThreshold;

  if (isBigPR) {
    warn('Big PR, try to keep changes smaller if you can...');
  }
}
